### ISSUE TITLE
False positive for react/button-has-type

### ISSUE THREAD

**Issue by remcohaszing** [yannickcr/eslint-plugin-react#1847]
The following code triggers `react/button-has-type`.

```js
document.createElement('button')
```

The culprit is in this function: https://github.com/yannickcr/eslint-plugin-react/blob/master/lib/rules/button-has-type.js#L15..L20

This should also check for the `pragma` setting.
